File: backend/schemas/blogs.py
from pydantic import BaseModel, Field


class BlogCreateRequest(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    excerpt: str | None = None
    content: str | None = None
    image: str | None = None
    category: str | None = Field(default=None, max_length=255)
    author: str | None = Field(default=None, max_length=255)


class BlogUpdateRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    excerpt: str | None = None
    content: str | None = None
    image: str | None = None
    category: str | None = Field(default=None, max_length=255)
    author: str | None = Field(default=None, max_length=255)

File: frontend/src/api/blogs.ts
import client from './client';
import type { Blog, BlogsResponse } from '../types';

export const fetchBlogs = async (): Promise<BlogsResponse> => {
  const { data } = await client.get<BlogsResponse>('/api/blogs');
  return data;
};

export const fetchBlog = async (blogId: number): Promise<Blog> => {
  const { data } = await client.get<Blog>(`/api/blogs/${blogId}`);
  return data;
};

// Admin
export const fetchAdminBlogs = async (): Promise<BlogsResponse> => {
  const { data } = await client.get<BlogsResponse>('/api/admin/blogs');
  return data;
};

export const createAdminBlog = async (payload: {
  title: string;
  excerpt?: string;
  content?: string;
  image?: string;
  category?: string;
  author?: string;
}): Promise<Blog> => {
  const { data } = await client.post<Blog>('/api/admin/blogs', payload);
  return data;
};

export const updateAdminBlog = async (
  blogId: number,
  payload: Partial<{
    title: string;
    excerpt: string;
    content: string;
    image: string;
    category: string;
    author: string;
  }>
): Promise<Blog> => {
  const { data } = await client.put<Blog>(`/api/admin/blogs/${blogId}`, payload);
  return data;
};

export const deleteAdminBlog = async (blogId: number) => {
  const { data } = await client.delete(`/api/admin/blogs/${blogId}`);
  return data;
};
